apacheconfig/wloader.py

```diff
@@ -0,0 +1,160 @@
+
+
+def _restore_original(word):
+    """If the `word` is a Quoted string, restores it to original.
+    """
+    if getattr(word, 'is_single_quoted', False):
+        return "'%s'" % word
+    if getattr(word, 'is_double_quoted', False):
+        return '"%s"' % word
+    return word
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Node():
+    """Generic class containing data that represents a node in the config AST.
+    """
+
+    @abc.abstractmethod
+    def __str__(self):
+        raise NotImplementedError()
+
+    @property
+    def type(self):
+        """A typestring as defined by the apacheconfig parser.
+
+        A single *Node class can have multiple possible values for this. For
+        instance, ItemNode is a generic representation of any single directive
+        but has a couple different possible `type`s, including `comment`,
+        `statement`, and `include`-- depending on which the caller may
+        decide to treat ItemNode differently.
+        """
+        if self._type is None:
+            raise NotImplementedError()
+        return self._type
+
+    @property
+    def whitespace(self):
+        """Trailing or preceding whitespace.
+
+        Each Item or BlockNode keeps track of the whitespace preceding it.
+        For the first element in the configuration file, there could be no
+        whitespace preceding it at all, in which case this should return the
+        empty string.
+
+        ContentsNode is special in that it keeps track of the trailing white-
+        space.
+        """
+        if self._whitespace is None:
+            raise NotImplementedError()
+        return self._whitespace
+
+    @whitespace.setter
+    def whitespace(self, value):
+        if self._whitespace is None:
+            raise NotImplementedError()
+        self._whitespace = value
+
+
+class ItemNode(Node):
+    """Contains data for a single configuration line.
+
+    Also manages any preceding whitespace. Can represent a key/value option,
+    a comment, or an include/includeoptional directive.
+
+    Construct this using raw AST data from parser. Generally, block data looks
+    like:
+        ['block', <open tag>, <contents object>
+    Examples of what raw AST data might look like for an ItemNode:
+        ['statement', 'option', ' ', 'value']
+        ['include', 'include', ' ', 'relative/path/*']
+        ['statement', '\n  ', 'option', ' = ', 'value']
+        ['comment', '# here is a comment']
+        ['comment', '\n  ', '# here is a comment']
+
+    Properties:
+        value: getter & setter for the final element in the raw AST. For
+               includes, this would be the path for include directives; for
+               comments, it's the comment body, and for key/value directives,
+               it's the value.
+        name:  getter only. Retrieves the first non-whitespace element in the
+               raw AST.
+    """
+
+    def __init__(self, raw):
+        """Initializes ItemNode with raw data from AST
```

Comment:
Arguably, putting `__init__` docstring into class docstring makes it looking better in Sphinx.